include/README.md

```diff
@@ -0,0 +1,399 @@
+# Basic Graph Library Concepts and low-level API
+
+The following is a brief introduction to the library architecture for people already vaguely familiar with GNU Radio,
+graph-based algorithms, or those simply looking for a quick-intro intro of the concepts and the 'where-is-what
+implemented'.
+For the 'how-it-is-implemented' please consult the provided links, code-references, and feel welcome to make PRs to
+continuously improve this document.
+
+## Overall Architecture
+
+![graph_core_structure_diagram_dark](./assets/graph_core_structure_dark.png#gh-dark-mode-only)
+![graph_core_structure_diagram_light](./assets/graph_core_structure.svg#gh-light-mode-only)
```

Comment:
Some questions/clarifications/suggestions:
- image differentiates between source/node/sink, the code doesn't. At the same time the domain argument looks like a template parameter, making it look a bit like code.
- data converters are visualized as blocks, which side (or both) call their work function or are they not really blocks? maybe add a bullet point for data_converter below?
- I'm not really sure I'm interpreting the lower(scheduler) part of the image correctly:
    - what do the arrows represent (calls/dataflow/?) Maybe add a verb to arrows or have different arrows for different semantics.
    - subgraph3 which is not present in the flowgraph below the nodes while the gpu scheduler is below the main scheduler, between the blocks and the compute.
    - I would expect the main scheduler to first call source1 (all other blocks seem to be topologically ordered)
    - subgraph3 which is not present in the flowgraph below the nodes while the gpu scheduler is below the main scheduler, between the blocks and the compute. (I know, it's optional, but I have the feeling that if it was in the graph I could maybe understand it better. Maybe having it greyed out in both the flow- and scheduler-graph would make it easier to understand, without overloading the image?)

Reply:
> At the same time the domain argument looks like a template parameter, making it look a bit like code.

This will (eventually) be a node NTTP arguement that user can specialise to implementa block that works on the CPU, GPU, etc...

Reply:
> data converters are visualized as blocks

Yes, they are blocks/node whose `work()` function needs to be explicitly called by the scheduler.

Reply:
Re: subgraph3

Yes, this was in the plot (grouping blocks #2 and #6), but seems to cut into the buffer representation and overall overloaded the picture a bit.

Regarding scheduling: we have some more graph topologies that are more usefull to explain the topology and ordering of the schedule... will add this later to the section `Collection of Ideas and known Scheduling Algorithms` also explaining why there cannot be a one-fits-all scheduler. This would also require gant-type charts that one would see in a profiler etc.... requires a bit more graphical design.

For this PR and picture this would IMO a bit too much.

Reply:
all fine, figures are always some kind of tradeoff between completeness and complexity, i guess. I was just trying to document the points where I felt that I first got it wrong or took longer to understand.

> >data converters are visualized as blocks
>
> Yes, they are blocks/node whose work() function needs to be explicitly called by the scheduler.

ok that makes sense, so the domain boundary would be at one of its ports and the implementer of the converters can decide to which flowgraph it belongs to, in the shown example, the cpu/main scheduler would transfer the data to the gpu buffer so the data_converter would be a `node<CPU,cpu_buffer_port, gpu_buffer_port>`.